projects/experimental/directives/pin/pin.directive.ts

```diff
@@ -0,0 +1,43 @@
+import {
+    ChangeDetectionStrategy,
+    Component,
+    Directive,
+    EventEmitter,
+    Inject,
+    Input,
+    Output,
+    ViewEncapsulation,
+} from '@angular/core';
+import {TuiDirectiveStylesService} from '@taiga-ui/cdk';
+
+@Component({
+    template: '',
+    styleUrls: ['./pin.styles.less'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    host: {
+        class: 'tui-pin',
+    },
+})
+export class TuiPinComponent {}
+
+@Directive({
+    selector: '[tuiPin]',
+    host: {
+        tuiPin: '',
+        '[class._open]': 'open',
+    },
+})
+export class TuiPinDirective {
+    @Input('tuiPin')
+    open: boolean | '' = false;
+
+    @Output()
+    readonly tuiPinChange = new EventEmitter<boolean>();
```

Comment:
Let's remove the output for now, we can always add it later.

Reply:
fixed